core/src/types/token.rs

```diff
@@ -847,6 +887,98 @@ pub fn minted_balance_key(token_addr: &Address) -> Key {
         .expect("Cannot obtain a storage key")
 }
 
+/// Obtain the nominal proportional key for the given token
+pub fn masp_kp_gain(token_addr: &Address) -> Key {
```

Comment:
Just a nit, but can this and the related functions below that yield a `Key` have "key" appended to their name, i.e. `masp_kp_gain_key`?